cupy/_core/_reduction.pyx

```diff
@@ -737,16 +741,13 @@ cdef class ReductionKernel(_AbstractReductionKernel):
 
         out_args = list(args[self.nin:])
         if out is not None:
-            if self.nout != 1:
-                raise NotImplementedError('')
             if len(out_args) != 0:
                 raise ValueError("cannot specify 'out' as both "
                                  "a positional and keyword argument")
-            out_args = [out]
 
         dev_id = device.get_device_id()
         in_args = _preprocess_args(dev_id, args[:self.nin], False)
-        out_args = _preprocess_args(dev_id, out_args, False)
```

Comment:
Let's keep `out_args` as `__call__` method also can take output arrays in positional arguments `args`. Instead, please assign `out` parameter to `out_args` with properly adjusted above.